packages/deployments/contracts/contracts/core/promise/libraries/PromiseMessage.sol

```diff
@@ -138,8 +138,8 @@ library PromiseMessage {
     }
     uint256 _length = lengthOfReturnData(_view);
     // before = 1 byte identifier + 32 bytes transferId + 20 bytes callback address + 1 byte success + 32 bytes length + x bytes data
-    // nonzero return data
-    return _length > 0 && (RETURNDATA_START + _length) == _len;
+    // allow zero-length return data
+    return _length >= 0 && (RETURNDATA_START + _length) == _len;
```

Comment:
I'm wondering why this was here before... copy/paste error, or just wrong impression when it was written at the time? If it's the former, we should make sure it's not an issue elsewhere right?

Reply:
nvm, just the one spot. interesting that we glossed over this!